Automattic/woocommerce-payments
# Update international IP address rule card behavior when the rule is being affected by another configuration

Fixes #6320 

#### Changes proposed in this Pull Request





This PR modifies the behavior and display of the international IP address rule card if the rule is being affected by the WooCommerce core selling locations general option. Previously, if the WooCommerce core selling locations option was set to 'sell to all countries' we'd show a notice at the bottom of the card but still display the toggle to enable the rule as shown in the screenshot below:

![275852485-4e80f817-04e9-415a-ab7a-4baa484daf8f](https://github.com/Automattic/woocommerce-payments/assets/60988591/d446d6c3-2133-47d3-ab25-f6c9354b1564)

The changes in this PR have updated the rule card display to not show the rule toggle and instead show the notice in the toggle's place with updated text indicating that filter is disabled because the WooCommerce core selling locations option is set to 'sell to all countries'. See the screenshot below:

![Screenshot on 2023-12-04 at 11_35_30](https://github.com/Automattic/woocommerce-payments/assets/60988591/3dd7a533-1b0e-49c7-8eef-c2cdb858bbeb)

The behavior and display of the rule card has not been changed if the the WooCommerce core selling locations option is set to the 'specific countries' or 'all countries except' options.





#### Testing instructions





* Check out this branch.
* All tests should pass.
* Go to WooCommerce > Settings
* Select 'Sell to all countries' in the Selling location(s) dropdown.
* Save.
* Go to Payments > Settings
* Scroll down to the fraud protection section, click advanced, then click configure.
* Scroll down to the International IP Address card and confirm that it looks like the second screenshot above. 
* Next, bring up your Jurassic Tube tunnel and confirm that the card looks okay on mobile.
* Go back to WooCommerce > Settings
* Select 'Sell to specific countries' in the Selling location(s) dropdown.
* Add a country or two to the 'Sell to specific countries' field if needed.
* Save.
* Scroll down to the fraud protection section, click advanced, then click configure.
* Scroll down to the International IP Address card and confirm that the toggle is present and that you have an info notice stating that orders from outside your specific countries will be blocked by the filter.
* Go back to WooCommerce > Settings
* Select 'Sell to all countries, except for...' in the Selling location(s) dropdown.
* Add a country or two to the 'Sell to all countries, except for...' field if needed.
* Save.
* Scroll down to the fraud protection section, click advanced, then click configure.
* Scroll down to the International IP Address card and confirm that the toggle is present and that you have an info notice stating that orders from the following countries will be blocked by the filter.

-------------------


**Post merge**
